### ISSUE TITLE
lag on `todo`; are we eagerly fetching historical names for things that aren't going to be displayed anyway?

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#1029]
We can maybe ask @seagreen for a copy of his hardtack repo if the problem isn't immediately obvious from one we make up.

**Comment by mitchellwrosen** [comment]
Here's the slow line: https://github.com/unisonweb/unison/blob/67254cc36f17c4715750640d004ae4b5a9cf410b/parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs#L2190

**Comment by pchiusano** [comment]
That’s helpful thanks @mitchellwrosen 

Fetching historical names should be fast for things in the recent past, but the slowness we were seeing seemed like a full history search.

**Comment by aryairani** [comment]
possibly related to #781

**Comment by aryairani** [comment]
I suppose the very idea of the historical search is flawed (because even in cases where we actually need to search through the whole history, we don't want to); however, maybe we can identify something that will help in this specific case.

**Comment by aryairani** [comment]
@mitchellwrosen Assuming that `Branch.findHistoricalRefs` isn't broken (and I'm guessing it isn't), then either 
1. `deps` includes something that it ought not include, and we should figure out how to remove the thing, or
2. everything in `deps` does need to be there, and the search is just gonna be slow.

The ticket topic supposes it's #1, in which case we need a more appropriate value for `deps`.

**Comment by mitchellwrosen** [comment]
@aryairani Are you able to reproduce this in any of your namespaces? If not I would be happy to publish mine somewhere for you to poke at

**Comment by aryairani** [comment]
@mitchellwrosen No, I can't reproduce in mine.  If you could publish yours, that would be great.

**Comment by mitchellwrosen** [comment]
@aryairani Here ya go https://github.com/mitchellwrosen/unison-slow-todo-branch

**Comment by runarorama** [comment]
This is coming up in testing the PR workflow. Running `todo` takes literally minutes, likely related to #781
